docs/config_to_support_additional_workspace_directories.md

```diff
@@ -0,0 +1,101 @@
+## Config to support additional workspace directories
+
+### Issue
+https://github.com/runfinch/finch/issues/107
+
+### Approach
+
+Lima yaml file has Mounts field to handle the mount points. Similar to other Finch config like cpu and memory, it is straightforward to write additional_workspace_directories to Lima’s override.yaml file.
+
+For example, for Finch config:
+```
+cpus: 2
+memory: 4GiB
+additional_workspace_directories:
+- /Volumes/mydir
+```
+
+Finch config applier will make override.yaml like this:
+
+```
+images: []
+cpus: 2
+memory: 4GiB
+mounts:
+- location: /Volumes/mydir
+mountPoint: /Volumes/mydir
+writable: true
+sshfs:
+cache: true
+followSymlinks: false
+sftpDriver: ""
+9p:
+securityModel: none
+protocolVersion: 9p2000.L
+msize: 128KiB
+cache: mmap
+networks:
+- lima: finch-shared
+```
+
+Different to cpu and memory, the “mounts” field in override.yaml will be appended to the default mounts instead of replacing it. So we don’t have to add the default home directory to the override.yaml file. Reference (https://github.com/lima-vm/lima/blob/master/pkg/limayaml/defaults.go#L410)
+
+### Error handling consideration
+
+There are lots of error cases if users directly pass random locations to Lima. For example, when users try to set “/” as in additional workspace directories.
+
+*Option 1 Directly pass the Lima and return whatever error Lima returns. Contribute to Lima to enhance error handling if needed. (Recommended)*
+
+Pros:
+
+* Easy to implement
+
+Cons:
+
+* Sometimes the error log in Lima context may not be descriptive in Finch context
```

Comment:
We'd probably just need to improve Lima's error handling to mitigate this, right? Do you have any examples of a "bad error" that Lima would currently emit, and any ideas on how to fix it?

Reply:
When I try to mount `/`, I get

```
FATA[0031] the YAML is invalid, saved the buffer as "lima.REJECTED.yaml": field `mounts[2].location` must not be a system path such as /etc or /usr
```

Which doesn't make sense if the user is only aware of `finch.yaml`

Reply:
I noticed in vm init and start, we already wrap Lima error messages with Finch error messages so should be sufficient for now. Let me know if you think this should be more descriptive.